vivo-project/Vitro
# Faux data properties and fixes for current faux properties view

# What does this pull request do?

- Fixes faux object properties view: now if faux property exists with the same domain, then real property is shadowed and duplicate faux property is shown instead. 
- Extended domain and range options for faux properties to all domains except not disjoint ones.
- Implemented faux properties for data proerties.

# What's new?

In PropertyTemplateModel removed ambiguous check that hidden real object properties.
FauxPropertyDaoJena, ListFauxPropertiesController, siteAdmin-fauxPropertiesList.ftl, dataprops_edit.jsp  modified to support faux data properties
Added code to get not disjoint classes in FormUtils
Created FauxDataPropertyWrapper.java to wrap faux data properties in GroupedPropertyList like it is done in FauxObjectPropertyWrapper for faux object properties
Refactored GroupedPropertyList and added new logic to support faux data properties and correctly list faux object properties
Created an interface FauxPropertyWrapper interface for faux property wrappers.
Refactored and extended ApplicationConfigurationOntologyUtils to get information about populated, not populated object and data faux properties.
Log exceptions in VClassDaoJena
Removed unused import in ObjectProperty 
Added safety check to [PropertyDaoJena.java](https://github.com/vivo-project/Vitro/compare/i18n-redesign...litvinovg:Vitro:faux_properties?expand=1#diff-332b248b5f5e7ea5dcfdb68fab7b467a8712df0e19b66c8a233b508b2c2578ce)
If possible I suggest to deprecate or remove all code from PropertyDaoJena that mixes real and faux properties from [this line](https://github.com/litvinovg/Vitro/blob/65a6ab48f1865829f99f84ce7919517d014a7fe3/api/src/main/java/edu/cornell/mannlib/vitro/webapp/dao/jena/PropertyDaoJena.java#L778). 

# How should this be tested?
Create data and object faux properties, add values. Test that faux properties are available even if domain of base property doesn't match individual types, but domain of faux property match individual type.
Faux data property creation should be available from data property page
Both types of faux properties listings should work.
Faux data properties range should always be the same as base property.
Faux data and object property ranges should be constrained by not disjoint properties.
This change require documentation to be updated

# Interested parties
@bkampe  @brianjlowe @chenejac
